controller/argument_value_resolver.rst

```diff
@@ -51,12 +55,12 @@ controller::
 
     namespace AppBundle\Controller;
 
-    use AppBundle\Entity\User;
     use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Security\Core\User\UserInterface;
 
     class UserController
     {
-        public function indexAction(User $user)
+        public function indexAction(UserInterface $user)
```

Comment:
I'm trying to understand.
ParamConverters are executed first. 
ArgumentValueResolvers are executed later.
They both allow you to type-hint arguments in your controller.
 When should you use a ParamConverter and when would you use an ArgumentValueResolver?